FRC1076-2016-Competition/test/org/usfirst/frc/team1076/test/controller/DistanceAutonomousTest.java

```diff
@@ -73,13 +70,15 @@ public void testVaribleSpeed() {
 	
 	@Test
 	public void testNoArmMotion() {
-		AutoState auto = new DistanceAutonomous(100, 10);
+		encoder.reset();
+		AutoState auto = new DistanceAutonomous(100, 10, encoder);
 		assertEquals(0, auto.armSpeed(), EPSILON);
 	}
 	
 	@Test
 	public void testNoIntakeMotion() {
-		AutoState auto = new DistanceAutonomous(100, 10);
+		encoder.reset();
+		AutoState auto = new DistanceAutonomous(100, 10, encoder);
 		assertEquals(0, auto.intakeSpeed(), EPSILON);
 	}
```

Comment:
Once you change it so it takes the "zero point" of the encoder into account, you should add a test that ensures that running two ones back to back don't cause a problem. For instance, move 1.5 units, then move 1.5 units again. The current implementation would have the second one stop instantly.